File: Tarea BubbleSort/Tarea5.java
import java.util.Random;

public class Tarea5 {
    public static void bubbleSort(int[] arr) {
        int n = arr.length;
        boolean intercambio;
        for (int i = 0; i < n - 1; i++) {
            intercambio = false;
            for (int j = 0; j < n - 1 - i; j++) {
                if (arr[j] > arr[j + 1]) {
                    int temp = arr[j];
                    arr[j] = arr[j + 1];
                    arr[j + 1] = temp;
                    intercambio = true;
                }
            }
            if (!intercambio) break;
        }
    }

    public static void main(String[] args) {
        Random rand = new Random();

        int[] arr = new int[10];
        for (int i = 0; i < arr.length; i++) {
            arr[i] = rand.nextInt(100) + 1;
        }

        System.out.print("Arreglo original: ");
        for (int num : arr) System.out.print(num + " ");
        System.out.println();

        bubbleSort(arr);

        System.out.print("Arreglo ordenado: ");
        for (int num : arr) System.out.print(num + " ");
    }
}

File: Tarea BubbleSort/Tarea5.py
import random
def bubblesort(arreglo):
    n = len(arreglo)
    for i in range(n):
        intercambio = False
        for j in range(0, n - i - 1):
            if arreglo[j] > arreglo[j + 1]:
                arreglo[j], arreglo[j + 1] = arreglo[j + 1], arreglo[j]
                intercambio = True
        if not intercambio:
            break       
arreglo = [random.randint(1, 100) for _ in range(10)]

print("Arreglo original:", arreglo)
bubblesort(arreglo)
print("Arreglo ordenado:", arreglo)
